packages/eslint-plugin-lit-a11y/lib/utils/getContextSourceCode.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns checks backward compatiblity and returns context.sourceCode
+ *
+ */
+function getContextSourceCode(context) {
+  try {
+    if (context.getSourceCode()) {
+      return context.getSourceCode();
+    }
+  } catch (e) {
+    return context.sourceCode;
+  }
+}
```

Comment:
Maybe we can add some more context on why this is needed and when we can remove it? I would also like to not throw and catch errors if possible.

```suggestion
/**
 * A backwards compatible way to get the source code of a given context.
 */
function getContextSourceCode(context) {
  return context.sourceCode ?? context.getSourceCode();
}
```

Is this now small enough to just inline into each of the rules? What do you think?

Reply:
I had tried this before, it breaks the execution with error "Function does not exist" and hence I added it in a try catch. I have updated the description of the method

Reply:
Hmm, but since we are checking the `context.sourceCode` now first and falling back to `context.getSourceCode()` shouldn't that still work?

Reply:
Can we align the implementation with how we did it in eslint-plugin-ing-security? I think its similar to what kris suggests here, and Michel tested it with both eslint 8 and 9 so that also addresses @justinfagnani s drive-by comment